# 
# Submitted-By: Robert Schwebel, 2005-12-07
# Committed-By: Robert Schwebel, 2005-12-07
#
# Error: 
#
# if gcc -DHAVE_CONFIG_H -I. -I. -I.. -I../include -I../include  \
# ... -g -O2 -Wall -Wcast-qual -Wmissing-prototypes -Wpointer-arith \
# -Wshadow -Wstrict-prototypes -MT ntp_config.o -MD -MP -MF ".deps/ntp_config.Tpo" \
#  -c -o ntp_config.o `test -f 'ntp_config.c' || echo './'`ntp_config.c; \
# then mv -f ".deps/ntp_config.Tpo" ".deps/ntp_config.Po"; \
# else rm -f ".deps/ntp_config.Tpo"; exit 1; \
# fi
# ntp_config.c:79: error: `CONFIG_KEYS' undeclared here (not in a function)
#
# Description: 
#
# When compiling with NATIVE=1 it may happen that some system include file comes 
# before ntp_config.h and seems to undefine CONFIG_KEYS again. Reordering the 
# includes fixes this. 
#
# State:
# 
# not submitted to upstream
#

diff -urN ntp-4.2.0/ntpd/ntp_config.c ntp-4.2.0-config/ntpd/ntp_config.c
--- ntp-4.2.0/ntpd/ntp_config.c	2003-10-07 10:35:14.000000000 +0200
+++ ntp-4.2.0-config/ntpd/ntp_config.c	2005-12-07 09:11:16.000000000 +0100
@@ -9,15 +9,6 @@
 # include <netinfo/ni.h>
 #endif
 
-#include "ntpd.h"
-#include "ntp_io.h"
-#include "ntp_unixtime.h"
-#include "ntp_refclock.h"
-#include "ntp_filegen.h"
-#include "ntp_stdlib.h"
-#include "ntp_config.h"
-#include "ntp_cmdargs.h"
-
 #include <stdio.h>
 #include <ctype.h>
 #ifdef HAVE_SYS_PARAM_H
@@ -40,6 +31,15 @@
 
 #include <netdb.h>
 
+#include "ntpd.h"
+#include "ntp_io.h"
+#include "ntp_unixtime.h"
+#include "ntp_refclock.h"
+#include "ntp_filegen.h"
+#include "ntp_stdlib.h"
+#include "ntp_config.h"
+#include "ntp_cmdargs.h"
+
 extern int priority_done;
 
 /*
